io/src/main/scala/fs2/io/JavaInputOutputStream.scala

```diff
@@ -144,9 +144,9 @@ private[io] object JavaInputOutputStream {
 
       def go(acc:Array[Byte]):F[Int] = {
         F.flatMap(readOnce(acc,0,1,queue,dnState)) { read =>
-          if (read < 0) F.pure(read)
```

Comment:
I'd prefer the `< 0` check to handle `InputStream` implementations that return other negative values. Per your point, that would be contrary to the docs but I think the risk/reward here favors `< 0`.